pulp_smash/tests/rpm/api_v2/test_remove_unit.py

```diff
@@ -0,0 +1,283 @@
+        cls.cfg = config.get_config()
+        cls.resources = set()
+
+    @classmethod
+    def tearDownClass(cls):
+        """Delete created resources."""
+        client = api.Client(cls.cfg)
+        for resource in cls.resources:
+            client.delete(resource)
+
+
+class _CommonTestsMixin(object):
+    """Common tests for RemoveMissing* classes."""
+
+    def test_status_code(self):
+        """Verify th HTTP status code of each server response."""
+        for step, code in (
+                ('sync', 202),
+                ('publish', 202),
+                ('units before removal', 200),
+                ('units after removal', 200),
+        ):
+            with self.subTest(step=step):
+                for response in self.responses[step]:
+                    self.assertEqual(response.status_code, code)
+        for step, code in (
+                ('distribute', 201),
+                ('remove unit', 202),
+        ):
+            with self.subTest(step=step):
+                self.assertEqual(self.responses[step].status_code, code)
+
+    def test_task_error_traceback(self):
+        """Assert each task's "error" and "traceback" fields are null."""
+        for action in {'sync', 'remove unit'}:
+            for i, task in enumerate(self.task_bodies[action]):
+                for key in {'error', 'traceback'}:
+                    with self.subTest((i, key)):
+                        self.assertIsNone(task[key])
+
+    def test_task_progress_report(self):
+        """Assert no task's progress report contains error details."""
+        for i, task in enumerate(self.task_bodies['sync']):
+            with self.subTest(i=i):
+                self.assertEqual(
+                    task['progress_report']['yum_importer']['content']['error_details'],  # noqa pylint:disable=line-too-long
+                    []
+                )
+
+    def test_units_before_removal(self):
+        """Test that units in repositories before removal are the same."""
+        bodies = [re.json() for re in self.responses['units before removal']]
+        # Package category and package group will differ so we count only RPMs
+        self.assertEqual(
+            set(unit['unit_id'] for unit in bodies[0]
+                if unit['unit_type_id'] == 'rpm'),  # This test is fragile
+            set(unit['unit_id'] for unit in bodies[1]
+                if unit['unit_type_id'] == 'rpm'),  # due to hard-coded
+        )  # indices. But the data is complex, and this makes things simpler.
+
+    def test_unit_removed(self):
+        """Test that correct unit from first repository has been removed."""
+        body = self.responses['units after removal'][0].json()
+        units_names = set(unit['metadata']['name'] for unit in body
+                          if unit['unit_type_id'] == 'rpm')
+        self.assertNotIn(self.removed_unit, units_names)
+
+
+class _RemoveMissingTestCase(_BaseTestCase):
```

Comment:
`_BaseTestCase` and `_RemoveMissingTestCase` can be collapsed into a single class. I don't see any advantage to splitting them up.
![classes](https://cloud.githubusercontent.com/assets/1012989/12955843/f3578b20-cff2-11e5-9d15-95f1423f5390.png)